api/collections/serializers.py

```diff
@@ -94,3 +96,72 @@ def get_absolute_url(self, obj):
                 'node_link_id': obj._id
             }
         )
+
+
+class LinkedNode(JSONAPIRelationshipSerializer):
+    class Meta:
+        type_ = 'linked_nodes'
+
+
+class CollectionLinkedNodesRelationshipSerializer(ser.Serializer):
+
+    data = ser.ListField(child=LinkedNode())
+    links = LinksField({'self': 'get_self_url',
+                        'html': 'get_related_url'})
+
+    def get_self_url(self, obj):
+        return obj['self'].linked_nodes_self_url
+
+    def get_related_url(self, obj):
+        return obj['self'].linked_nodes_related_url
+
+    class Meta:
+        type_ = 'linked_nodes'
+
+    def get_pointers_to_add_remove(self, pointers, new_pointers):
+        diff = relationship_diff(
+            current_items={pointer.node._id: pointer for pointer in pointers},
+            new_items={val['node']['_id']: val for val in new_pointers}
+        )
+
+        nodes_to_add = []
+        for node_id in diff['add'].keys():
+            node = Node.load(node_id)
+            if not node:
+                raise exceptions.NotFound
+            nodes_to_add.append(node)
+
+        return nodes_to_add, diff['remove'].values()
+
+    def make_instance_obj(self, obj):
+        # Convenience method to format instance based on view's get_object
+        return {'data': [
+            pointer for pointer in
+            obj.nodes_pointer
+            if not pointer.node.is_deleted and not pointer.node.is_folder
+        ], 'self': obj}
+
+    def update(self, instance, validated_data):
+        collection = instance['self']
+        auth = get_user_auth(self.context['request'])
+
+        add, remove = self.get_pointers_to_add_remove(pointers=instance['data'], new_pointers=validated_data['data'])
+
+        for pointer in remove:
+            collection.rm_pointer(pointer, auth)
```

Comment:
OK, it would be good to have tests verifying this. I don't, for example, see a test for the client passing a node ID that is already part of the collection (should return a 400).